### ISSUE TITLE
daps 2.3.0: Validation error message <xml=version="1.0"... with "daps-init" example

### ISSUE THREAD

**Issue by LoSong** [openSUSE/daps#386]
## Problem description
Validation of example book file causes error message and aborts validation.

## Actual behavior
`daps -d /home/user/path/to/MAIN-File.xml validate` causes the following error message:

```
/usr/bin/daps: Zeile 1138: declare: '<?xml=version="1.0" encoding="UTF-8': Ist kein gültiger Bezeichner
ERROR: Invalid value in config file /home/user/path/to/MAIN-File.xml in line 1
```

Of course, `<?xml=version=...` is wrong, but I don't know, where daps gets this from.
The MAIN-File.xml correctly contains
`<?xml version="1.0" encoding="UTF-8"?>`

## Steps to reproduce the behavior

- Create demo book file with
`daps-init --docdir="/home/user/path/to/" --docbook5 --name=File --rootelement=book`

- file and directory structure are created correctly

- Validate demo book file with
`daps -d /home/user/path/to/MAIN-File.xml validate`

The above error message appears. Consequently a transformation into PDF etc. fails.

daps: 2.3.0
OS: Linux Mint 18 Cinnamon 64-bit
Cinnamon Version: 3.0.7
Linux-Kernel: 4.4.0-47-generic

**Comment by fsundermeyer** [comment]
Works for me on openSUSE, so I assume it is a Linux Mint compatibility issue. Looks like I need to install Mint and test it...

Could you post the output of

`daps -d /home/user/path/to/MAIN-File.xml --debug validate`

here? That will hopefully help me to narrow down the issue...

**Comment by LoSong** [comment]
Hi,

I would post it here, but there is no debug output... just again the error message I've posted above. I'm confused.

Regards from Aachen

**Comment by fsundermeyer** [comment]
The very first command daps runs before doing anything else is

`xmllint --nonet --noout --nowarning --xinclude --loaddtd /home/user/path/to/MAIN-File.xml'

It checks whether the XML file is well-formed. Looks like this command already fails for you (for whatever reason). What happens if you run the command as posted above (with the pacth to the XML file adjusted, of course)? It should not output anything.

**Comment by LoSong** [comment]
Yes, no output.

Parallel to this I'm downloading OpenSuse to test it. openSUSE Leap 42.2 works with daps?

**Comment by LoSong** [comment]
Ok, no problem so far with OpenSuse 13.1.

From now on I'm using this for working with DocBook instead of Linux Mint.

May I suggest implementing an option "--debugstep" to make daps wait for the users confirmation for every command it executes?

**Comment by ghost** [comment]
@LoSong Note that 13.1 is positively ancient and not really supported at this point -- if you can update, use Leap 42.1 or Leap 42.2.

**Comment by LoSong** [comment]
Ok, updated to Leap 42.2, and it works very well.

Thanks!

**Comment by ghost** [comment]
@LoSong Is this problem now fixed? Should we close this?

**Comment by LoSong** [comment]
Hi,

I didn't test it again with Linux Mint, because it works with Leap 42.2.

So maybe there still is an issue with Mint... the recommendation for using OpenSuse (or avoidance of Linux Mint) could do the trick.

Regards.

**Comment by ghost** [comment]
Aah right, sorry. I just read enough of the thread to see that you solved your problem but not enough to see what the original problem was...

**Comment by fsundermeyer** [comment]
We should keep this one open to test DAPS on LInux Mint. THis is most likely just an installation or dependency issue.

**Comment by fsundermeyer** [comment]
Finally found the time to reproduce and test on Linux Mint (18.1):

1. The original error was no bug but caused by using the XML file with -d rather than the DC-file

       WRONG:   daps -d /home/user/path/to/MAIN-File.xml validate
       CORRECT: daps -d /home/user/path/to/DC-File.xml validate

2. The error message caused by the wrong command was not helpful. This has been corrected with 4c00125926eb45be3b26c169e392a25f7e917b0a
3. Found two minor bugs in daps-init and fixed them with 4a65b3e9ac0d0b636316046b6e2fe42fafc1ce87 and 876d70665b235b2e5f2402a58dc4ec3f2ce7b208
4. Successfully tested DAPS on Linux Mint

The fixes will be published with the upcoming DAPS 2.4.
